src/swap.ts

```diff
@@ -14,11 +25,31 @@ export type SwapEffect = {
   price: number;
 };
 
+/**
+ * Swap class represents a swap trade if an amount of asset on a particular pool.
+ *
+ * The swap class contains methods to ensure the swap is valid and prepare the transaction. It also contains
+ * a method to report the effect of the swap on the current pool values.
+ */
 export class Swap {
+  /** The effect of the swap computed at the time of construction. */
   effect: SwapEffect;
 
+  /** The asset deposited in order to make the swap. */
```

Comment:
It uses the opposite naming. Probably a mistake from my end but it's risky to change it now as it might change user code in an unpredictable way.
`out` is what the user deposits in a contract.
`in` is what comes from the contract to the user.
Basically, the naming is from the user perspective, not the contract perspective.
The funds go **out** of the user account and **in**to the user account.